src/broadcaster/broadcaster.py

```diff
@@ -38,11 +38,15 @@ class TestableBroadcaster(Broadcaster):
     def __init__(self):
         self.data = None
 
-    def update_data(self, data=None):
+    def update_dictionary(self, data=None):
+        """ Stores dictionary into data, updating keys """
         if data is not None:
             self.data = data
+            for key in self.data.keys():
+                self.update_key(key=key)
 
-    def read_data(self, key=None):
+    def update_key(self, key=None):
+        """ Updates key, returns data[key] """
         if self.data is None or key is None or not key in self.data or self.data[key] is None:
```

Comment:
add some parentheses to clarify

Reply:
Removed; invalid key and null errors to be handled outside of broadcaster class.